src/share/classes/com/sun/javatest/regtest/report/XMLWriter.java

```diff
@@ -177,11 +178,14 @@ private void insertProperties() throws TestResult.Fault {
 
     private String getOutput(String name) throws TestResult.Fault {
         String[] titles = tr.getSectionTitles();
-        for (int i = 0; i < titles.length; i++) {
-            if (titles[i].equals("main")) {
-                Section s = tr.getSection(i);
-                for (String x : s.getOutputNames()) {
-                    return s.getOutput(name);
+        //first we are looking for a "main" section in jtr log.. if none is found it is usually cuz the test is executed via script, hence under "shell" section
+        for (String section : Arrays.asList("main", "shell")){
+            for (int i = 0; i < titles.length; i++) {
+                if (titles[i].equals(section)) {
+                    Section s = tr.getSection(i);
+                    for (String x : s.getOutputNames()) {
+                        return s.getOutput(name);
+                    }
```

Comment:
The outer loop seems weird. It would be simpler to just change

 `if (titles[i].equals(section))` 
 
 to 
 
 `if (titles[I].equals("main") || titles[I].equals("shell"))` 
 
 or equivalent

Reply:
or `if (Set.of("main", "shell").contains(titles[i]))` or equivalent

Reply:
fixed.. I did not know if there is any chance of jtharness producing both Main and Shell section in the log in some corner cases, so I decided to play it safe and made the Shell section as a fallback option when no Main section is found.

Reply:
FYI, a "section" contains the output for an "action" in the test description, so a test description containing a sequence or mix of `@run main` and `@run shell` actions will generate corresponding sections in the `.jtr` file.